src/codec/mod.rs

```diff
@@ -412,6 +450,53 @@ fn decode_response_pdu_bytes(bytes: Bytes) -> io::Result<Response> {
             }
             ReadWriteMultipleRegisters(data)
         }
+        0x2b if rdr.read_u8()? == MEI_TYPE_READ_DEVICE_IDENTIFICATION => {
```

Comment:
Using a side-effect within a match guard is both confusing and an error. The value is consumed and lost for the following match arms.